packages/aws-client/src/S3.ts

```diff
@@ -739,34 +740,32 @@ export const uploadS3FileStream = (
     },
   });
 
+export interface ListS3ObjectsOutput {
+  Key: string
+}
+
 /**
  * List the objects in an S3 bucket
- *
- * @param {string} bucket - The name of the bucket
- * @param {string} prefix - Only objects with keys starting with this prefix
- *   will be included (useful for searching folders in buckets, e.g., '/PDR')
- * @param {boolean} skipFolders - If true don't return objects that are folders
- *   (defaults to true)
- * @returns {Promise} A promise that resolves to the list of objects. Each S3
- *   object is represented as a JS object with the following attributes: `Key`,
- * `ETag`, `LastModified`, `Owner`, `Size`, `StorageClass`.
  */
 export const listS3Objects = async (
   bucket: string,
   prefix?: string,
   skipFolders: boolean = true
-) => {
+): Promise<Array<ListS3ObjectsOutput>> => {
   log.info(`Listing objects in s3://${bucket}`);
   const params: ListObjectsRequest = {
     Bucket: bucket,
   };
   if (prefix) params.Prefix = prefix;
 
   const data = await s3().listObjects(params);
-  let contents = data.Contents || [];
+  if (!data.Contents) {
+    return [];
+  }
+  let contents = data.Contents.filter((obj) => obj.Key !== undefined) as Array<ListS3ObjectsOutput>;
   if (skipFolders) {
     // Filter out any references to folders
-    contents = contents.filter((obj) => obj.Key !== undefined && !obj.Key.endsWith('/'));
+    contents = contents.filter((obj) => !obj.Key.endsWith('/'));
```

Comment:
I figure this should filter out undefined even if I haven't asked it to filter out directories